src/unix/process.c

```diff
@@ -372,11 +372,8 @@ static void uv__process_child_init(const uv_process_options_t* options,
     _exit(127);
   }
 
-  if (options->env != NULL) {
-    environ = options->env;
-  }
-
-  execvp(options->file, options->args);
+  execve(options->file, options->args, 
+         options->env != NULL ? options->env : environ);
```

Comment:
execve() doesn't search the PATH like execvp() does.  I realize it may be more POSIX-compliant but it would break a lot of existing code for mostly academic reasons.  We rely on the platform's libc to implement execvp() properly and they all do.

The other change looks fine to me.